src/Renci.SshNet/ShellStream.cs

```diff
@@ -568,10 +596,14 @@ public string Expect(Regex regex, TimeSpan timeout)
                         {
                             _incoming.Dequeue();
                         }
+                        
+                        text = text.Substring(0, match.Index + match.Length);
                         break;
                     }
                 }
 
+                if (_isDisposed) { return null; }
```

Comment:
Update the xmldoc for this method (and caller of this method) to mention that we now also return _null_ if the channel was closed while waiting for the expected expression.

Reply:
Ok!